src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/FileEntryCreator.kt

```diff
@@ -0,0 +1,54 @@
+package pl.edu.uj.ii.ksi.mordor.services
+
+import java.io.File
+import org.springframework.stereotype.Service
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileContent
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileEntry
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileMetadata
+import pl.edu.uj.ii.ksi.mordor.persistence.repositories.FileContentRepository
+import pl.edu.uj.ii.ksi.mordor.persistence.repositories.FileEntryRepository
+import pl.edu.uj.ii.ksi.mordor.persistence.repositories.FileMetadataRepository
+
+@Service
+class FileEntryCreator(
+    private val metadataExtractor: MetadataExtractor,
+    private val entryRepository: FileEntryRepository,
+    private val metadataRepository: FileMetadataRepository,
+    private val contentRepository: FileContentRepository,
+    private val fileTextExtractor: FileTextExtractor
+) {
+
+    fun create(file: File): FileEntry {
```

Comment:
rename to findOrCreate, try to first find a matching existing FileContent by hash, parse only if not found.

Reply:
FileEntry should be created even if metadata with the same hash exists, as it only maps a path to metadata. Therefore I think `create` is a better name for this function